### ISSUE TITLE
Missing backticks, missing PHP and argument passing in Wiki CRON page

### ISSUE THREAD

**Issue by galbaras** [PromInc/organic-search-analytics#64]
**Missing backtics**

The cron setup instructions show this:

0 1 * * * data-capture-run.php googleSearchAnalytics http://yourdomain.com/ date --date='4 days ago' '+%Y-%m-%d'

But should show this (date command wrapped in backticks):

0 1 * * * data-capture-run.php googleSearchAnalytics http://yourdomain.com/ \`date --date='4 days ago' '+%Y-%m-%d'\`

The backticks should be escaped with a backslash to display correctly.

**Missing PHP**

The command used by cron should start with "php". Otherwise, the system won't know what to do with the script. At least, it doesn't in my case (cPanel server). So the correct command line should be:

0 1 * * * php data-capture-run.php googleSearchAnalytics http://yourdomain.com/ \`date --date='4 days ago' '+%Y-%m-%d'\`

This assumes that the system "knows" where to find the php program. Otherwise, the full path should be used, which may vary.

**LiteSpeed section**

I'm using Organic Search Analytics on a cPanel/LiteSpeed server, and when the data capture script is run from cron, it passes the arguments via $_GET, not $argv, as you would expect. Therefore, I've had to change the command line to include argument names, like this:

0 1 * * * php data-capture-run.php type=googleSearchAnalytics domain=http://yourdomain.com/ date=\`date --date='4 days ago' '+%Y-%m-%d'\` | grep -i error

The ending "| grep -i error" stops the job from sending me emails on successful completion, which is what will happen normally, but produces output when there are errors, so that I will know about them.

**Comment by galbaras** [comment]
I run the data capture daily on a cPanel/LiteSpeed server. For each domain, I see these two warnings in error_log:

[12-Jun-2017 14:00:50 UTC] PHP Notice:  Undefined index: DOCUMENT_ROOT in <installation_dir>/inc/code/globalIncludes.php on line 4
[12-Jun-2017 14:00:50 UTC] PHP Notice:  Undefined index: SCRIPT_FILENAME in <installation_dir>/inc/code/globalIncludes.php on line 7

(In the log file, the actual installation directory appears instead of <installation_dir>).

This makes me think that the best way to run the cron job is using wget or cURL. This would make the job look like this:

0 1 * * * /usr/bin/curl "OSADOMAIN/data-capture-run.php?type=googleSearchAnalytics&domain=GSCDOMAIN&date=`date --date='4 days ago' '+%Y-%m-%d'`" | grep -i error

OSADOMAIN is the (local) domain, where Organic Search Analytics is installed. GSCDOMAIN is the domain being tracked on Google Search Console. The URL is wrapped in double quotes to ensure that cURL gets it unchanged.

By emulating a browser, everything should work well. I've tested it successfully, although having a URL as a parameter triggered a firewall rule, which wasn't fun. Maybe it's possible to use the site's reference from the database instead of its URL?

**Comment by PromInc** [comment]
**Backticks**
Thank you for catching that syntax error.  The wiki page has been updated.

**Path to Installation**
Good point - I've updated the wiki page for clarity.

**Missing PHP declaration**
Most Linux distributions will understand the script being requested by the cron job as being PHP and thus declaration should not be required.  I'm guessing that your need to declare it is a uniqueness to the LiteSpeed server you are using. (?)

**cPanel/LiteSpeed server**
I have no experience with this type of a server configuration.  I appreciate you sharing the solution you found to help others in the community that may be running on this configuration.

**error_log**
Interesting observation.  I'll need to retest a cron task on a traditional linux box when I have time here.  There is a chance that the `DOCUMENT_ROOT` and `SCRIPT_FILENAME` parameters are only passed on a HTTP request (as you suggest as a solution) and not via a direct PHP request.

That said, the usage of the variables these parameters are being accessed to populate are not used very heavily (at this time...) and are actually used primarily for loading external assets such as JS and CSS files on the report and settings pages.  So the fact that these errors are thrown when using a cron request really does not have a negative impact.

But I do feel error free code is valuable and ideal and thus looking into this issue deeper is something that will be done as it could have a growing impact in future versions.
